### PR TITLE
Run Travis & GH actions CI side by side

### PR DESCRIPTION
**Pull request by armenzg**
This runs most of what Travis executes inside of Github actions.
Running against the latest golang has not been ported. All other modes have been ported.

I will enquire as to how to get the latest golang running with the setup-go action.

Since I do not have write access on the upstream repo the checks won't show up on this PR.
You can view the results on my [fork](https://github.com/armenzg/sentry-go/actions/runs/481210918).

Related to https://github.com/getsentry/sentry-go/issues/313.

### CODE REVIEW

**Comment by armenzg** [comment]
Status checks running green: https://github.com/armenzg/sentry-go/actions/runs/481210918

**Comment by armenzg** [comment]
@rhcarvalho would you mind reviewing this?

**Review by billyvg — COMMENTED** [review]
LGTM, but will leave it up to repo owners for final approval

File: .github/workflows/ci.yml
Comment (armenzg) [inline .github/workflows/ci.yml:6]: We need to remove this before merging.

File: .github/workflows/ci.yml
Comment (armenzg) [inline .github/workflows/ci.yml:20]: This block adds `goflags` *only* for the `go111module: on` configuration.

File: .github/workflows/ci.yml
Comment (armenzg) [inline .github/workflows/ci.yml:23]: Is the `GOPATH` required for both modes?

File: .github/workflows/ci.yml
Comment (armenzg) [reply .github/workflows/ci.yml:20]: Would you be able to verify via the logs that `goflags` is being used properly for module mode on?

File: .github/workflows/ci.yml
Comment (armenzg) [inline .github/workflows/ci.yml:34]: Required for linting.

File: .github/workflows/ci.yml
Comment (armenzg) [inline .github/workflows/ci.yml:53]: Over here I decided that we don't need to run linting for both modes. Please correct me if incorrect assumption.

File: .github/workflows/ci.yml
Comment (armenzg) [inline .github/workflows/ci.yml:57]: Do you want this step? I can remove it.

Comment (rhcarvalho) [reply .github/workflows/ci.yml:57]: We don't need it.

File: .github/workflows/ci.yml
Comment (rhcarvalho) [reply .github/workflows/ci.yml:53]: 👍 

I think we've always run it because it did not have much impact on runtime, can't think now of a case where the assumption would be wrong.

File: .github/workflows/ci.yml
Comment (rhcarvalho) [reply .github/workflows/ci.yml:23]: Only important for `GO111MODULE=on`

File: .github/workflows/ci.yml
Comment (rhcarvalho) [reply .github/workflows/ci.yml:20]: Hard to tell with the current output. We can debug it with `go env` to print Go-related variables.

Comment (rhcarvalho) [reply .github/workflows/ci.yml:20]: If I understood https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/migrating-from-travis-ci-to-github-actions#using-environment-variables-in-a-matrix well, we need to set the env explicitly in the step(s) that we need it.

`go env` will tell us if we got it right.

File: .github/workflows/ci.yml
Comment (armenzg) [reply .github/workflows/ci.yml:23]: To my surprise, it is necessary for `go get` and `go build`.

It must be because of:
```
defaults:
      run:
        working-directory: ${{ github.workspace }}/src/github.com/getsentry/sentry-go
```

Are you OK if we leave `GOPATH` in place? Alternatively we can separate the flows for module on vs off to avoid all this custom steps.

**Comment by armenzg** [comment]
Latest results:
https://github.com/armenzg/sentry-go/actions/runs/483382799

We're postponing the following:
* Why GOPATH needs to be set for Modules off
* Testing against `tip`

I will tackle handling `tip` in a follow up PR.

**Comment by armenzg** [comment]
@rhcarvalho I also got `tip` working in this [other PR](https://github.com/armenzg/sentry-go/pull/1). Do you want me to include it in this PR? Or keep it separate?

**Comment by rhcarvalho** [comment]
Let's get this in and iterate on it in the next days as we see it running side-by-side with Travis. Then we drop Travis.

**Review by rhcarvalho — APPROVED** [review]

**Pull request merged**
